kratos/spatial_containers/bins_dynamic_objects.h

```diff
@@ -442,20 +442,22 @@ class BinsObjectDynamic {
    * @param MaxNumberOfResults [description]
    */
   void SearchObjectsInRadius(IteratorType const& ThisObjects, SizeType const& NumberOfObjects, std::vector<double>& Radius, std::vector<std::vector<PointerType> >& Results, std::vector<SizeType>& NumberOfResults, SizeType const& MaxNumberOfResults) {
-    PointType Low, High;
-    SearchStructureType Box;
 
-    #pragma omp parallel for private(Low,High,Box)
-    for(int i = 0; i < static_cast<int>(NumberOfObjects); i++) {
-      ResultIteratorType ResultsPointer            = Results[i].begin();
+    using tls_type = std::tuple<PointType, PointType, SearchStructureType>;
```

Comment:
It is the same, but for the sake of understanding i would propose doing it as

~~~cpp
class tls_type 
{
   public:
   PointType Low;
   PointType High;
   SearchStructureType Box;
}

ndexPartition<std::size_t>(NumberOfObjects).for_each(tls_type(), [&](std::size_t i, tls_type& rTLS){
        ResultIteratorType ResultsPointer = Results[i].begin();	        
        ResultIteratorType ResultsPointer = Results[i].begin();
        PointType& Low = rTLS.Low;
        PointType& High = rTLS.High;
        SearchStructureType& Box = rTLS.Box;

~~~

Reply:
ok
but I will use a `struct` :)

Reply:
ok, then you can omit the "public" keyword :-)

Reply:
done